From 3594b40102c3d0c67c05d69c0daa9387f72f2ef9 Mon Sep 17 00:00:00 2001
From: Du Chenyang <chenyang.du@intel.com>
Date: Mon, 16 Jul 2012 18:44:51 +0800
Subject: [PATCH] Adding build configuration for boot.bin

BZ: 46708

Adding an option to support intel boot.bin build, which will disable
the default build of boot.img if TARGET_MAKE_NO_DEFAULT_BOOTIMAGE is true.

Fix bootimage-nodeps target

BZ: 176055

also allow to override MKBOOTIMG (for fdk intel image)
and add a few target to the dontbother list,
to save developper time.

Change-Id: If4a85f757ca8eecce30542b9fd54e7652d1bed3f
Category: aosp improvement
Domain: AOSP-Framework-CoreService
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Sun Yijin <yijin.sun@intel.com>
Signed-off-by: Bruce Beare <bruce.j.beare@intel.com>
Signed-off-by: Du Chenyang <chenyang.du@intel.com>
Signed-off-by: Jocelyn Falempe <jocelyn.falempe@intel.com>
---
 core/Makefile |   13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/core/Makefile b/core/Makefile
index 45c318f..56b2710 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -477,6 +477,7 @@ ifdef BOARD_KERNEL_PAGESIZE
   INTERNAL_BOOTIMAGE_ARGS += --pagesize $(BOARD_KERNEL_PAGESIZE)
 endif
 
+ifneq ($(TARGET_MAKE_NO_DEFAULT_BOOTIMAGE),true)
 INSTALLED_BOOTIMAGE_TARGET := $(PRODUCT_OUT)/boot.img
 
 ifeq ($(TARGET_BOOTIMAGE_USE_EXT2),true)
@@ -522,6 +523,18 @@ bootimage-nodeps: $(MKBOOTIMG)
 
 endif # TARGET_BOOTIMAGE_USE_EXT2
 
+else # TARGET_MAKE_NO_DEFAULT_BOOTIMAGE
+
+$(INSTALLED_BOOTIMAGE_TARGET): $(MAKE_NO_DEFAULT_BOOTIMAGE_ITEMS)
+	$(hide) $(MAKE_NO_DEFAULT_BOOTIMAGE)
+
+.PHONY: bootimage-nodeps
+bootimage-nodeps:
+	@echo "make $@: ignoring dependencies"
+	$(hide) $(MAKE_NO_DEFAULT_BOOTIMAGE)
+
+endif # TARGET_MAKE_NO_DEFAULT_BOOTIMAGE
+
 else	# TARGET_NO_KERNEL
 # HACK: The top-level targets depend on the bootimage.  Not all targets
 # can produce a bootimage, though, and emulator targets need the ramdisk
-- 
1.7.10.4

